windowsforms/Sudoku/VB/sudoku/gui/Form1.vb

```diff
@@ -0,0 +1,84 @@
+Public Class Form1
+
+    Private WithEvents Game As New Game
+    Private ReadOnly r As New Random
```

Comment:
This will need to be removed if you updated the NewGame method as I mentioned earlier.

Reply:
I hate the use of WithEvents here, I just didn't want to change the code. Let me know if you want the logic changed. I also think the dropdown "Clue" is impossible to understand and the solution logic marks perfectly valid solutions as wrong.

Reply:
@paul1956 I wasn't mentioning the line that have WithEvents.
I meant:  `Private ReadOnly r As New Random`.
In one of the suggestions, I was suggesting that the NewGame method be parameterless, and it create the Random object internally.

Reply:
The Random generator is used in Form to show hints and in Game to create the game board so someone decided to create it once and pass it around. If also helps for testability.

Reply:
@paul1956, Didn't notice it's used in the form too. Then keep it unless someone came with a better design.